code/modules/plumbing/_the_tank.dm

```diff
@@ -0,0 +1,132 @@
+			use_power = 2
+			reagents.add_reagent_list(buffer.reagents.get_reagent_proportions())
+		if(PLUMBING_TANK_DRAIN)
+			use_power = 2
+			for(var/reagent in reagents.reagent_list)
+				var/datum/reagent/R = reagent
+				reagents.remove_reagent(R.id, 1)
+
+/obj/machinery/plumbing_tank/interact(mob/user)
+	var/dat
+	switch(obj_integrity / max_integrity)
+		if(0)
+			dat += "<i>The display lies dark, and the tank in ruins. Oops.</i>"
+		if(0 to 0.2)
+			dat += "<i>The entire tank shudders and heaves as it struggles to stay intact. A flurry of sparks spews from a cut wire.</i>"
+		if(0.2 to 0.4)
+			dat += "<i>[src] is dented, and badly. It could use some repairs.</i>"
+		if(0.4 to 0.6)
+			dat += "<i>[src]'s bolts are a little loose, and one of the gauges is cracked.</i>"
+		if(0.6 to 0.8)
+			dat += "<i>The tank's surface is a little scratched, and there's a good-sized dent near the top.</i>"
+		if(0.8 to 0.99)
+			dat += "<i>[src] seems to have suffered some minor damages, but nothing terrible.</i>"
+		if(0.99 to 1)
+			dat += "<i>[src] is running as well as can be expected.</i>"
+	dat += "<br><br>"
+	switch(use_power)
+		if(0)
+			dat += "<i>The power light is unlit.</i>"
+		if(1)
+			dat += "<font color='#00FF00'>The power light is blinking green.</font>"
+		if(2)
+			dat += "<font color='#00FF00'>The power light is glowing green.</font>"
+	if(powered())
+		dat += "<br><br><b>CNTS ([reagents.total_volume / reagents.maximum_volume]):</b><br>"
+		for(var/V in reagents.reagent_list)
+			var/datum/reagent/R = V
+			dat += "	reag. [R.id] vol. [R.volume]<br>"
+		dat += "<br>"
+		switch(operating_mode)
+			if(PLUMBING_TANK_IDLE)
+				dat += "<font color='#003FFF'>An LED labeled \"OPMD\" is glowing a steady blue.</font>"
+			if(PLUMBING_TANK_SYNTHESIZE)
+				dat += "<font color='#00FF00'>An LED labeled \"OPMD\" is flickering green[reagents.total_volume == reagents.maximum_volume ? "" : ", and you hear flowing liquid"].</font>"
+			if(PLUMBING_TANK_DRAIN)
+				dat += "<font color='#FF0000'>An LED labeled \"OPMD\" is blinking red, and you hear [reagents.total_volume ? "the sound of draining liquid" : "sucking sounds"].</font>"
+	dat += "<br><br>"
+	dat += "<a href='?src=\ref[src];switch_modes=1'>Set the OPMD dial to a new setting</a>"
+	var/datum/browser/popup = new(user, "plumbing_tank", name, 600, 400)
+	popup.set_content(dat)
+	popup.set_title_image(user.browse_rsc_icon(icon, icon_state))
+	popup.open()
+
+/obj/machinery/plumbing_tank/Topic(href, href_list)
+	if(..())
+		return
+	if(!iscarbon(usr) || !usr.canUseTopic(src) || !src || QDELETED(src) || stat)
+		return
+	var/mob/living/carbon/user = usr
+	if(href_list["switch_modes"])
+		var/new_mode = alert(user, "There are three options for the dial.", "OPMD Dial", "IDLE", "SYNT", "DRAI")
+		var/mode_fluff = ""
+		if(!user.canUseTopic(src) || !src || QDELETED(src) || stat)
+			return
+		switch(new_mode)
+			if("IDLE")
+				operating_mode = PLUMBING_TANK_IDLE
+			if("SYNT")
```

Comment:
They're restricted to four letters for fluff purposes. The display is meant to be obscure.